Checkout/Checkout/Source/Model/Card+Scheme.swift

```diff
@@ -111,19 +115,40 @@ extension Card {
       switch self {
       case .unknown:
         return nil
-      case .amex:
+      case .americanExpress:
         return 4
       case .visa,
         .mada,
         .mastercard,
         .maestro,
         .discover,
-        .diners,
+        .dinersClub,
         .jcb:
         return 3
       }
     }
 
+    /// maximum card length for any scheme
+    static let maxCardLengthAllSchemes = 19
+
+    var maxCardLength: Int? {
+      switch self {
+      case .unknown:
+        return nil
+      case .mada,
+        .visa,
+        .mastercard:
+        return 16
+      case .americanExpress:
+        return 15
+      case .maestro,
+        .discover,
+        .dinersClub,
+        .jcb:
+        return 19
```

Comment:
The following code, if `scheme.maxCardLength` is `nil` then use `Card.Scheme.maxCardLengthAllSchemes` 

```  
private func withinCardLengthLimit(_ cardNumber: String, for scheme: Card.Scheme) -> Bool {
    let maxCardLength = scheme.maxCardLength ?? Card.Scheme.maxCardLengthAllSchemes
    return cardNumber.count <= maxCardLength
 }
```
so i suggest, the max is always 19 

```suggestion
        return maxCardLengthAllSchemes
```